subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/UriTextResourceLoader.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.dsl.dependencies;
```

Comment:
This is sort of an odd package for this to be in.

What do you think about this being on `TextResourceFactory` (that may open other problems since it's a public API)?  Something like `TextResource fromUri(URI source)`.

That might not be right.... This is a different kind of `TextResource` apparently :/